### PR TITLE
batch_size related changes for Orca PyTorch

### PR DESCRIPTION
**Pull request by xiejinglei**
This PR relates to issue #3446 

Similar to `TF2`, after the changes, the program expects users to input `batch_size` in `fit`, `evaluate` and `predict` functions instead of in `config`. I also removed 
```py
if "batch_size" not in config:
```
in `torch_runner.py` because this may lead the program to ignore the input `batch_size`, for example, in the `validate` function.

In addition, the `data_creator` function should now expect 2 inputs: `config` and `batch_size`. Please have a look if the changes are ok. Thanks

### CODE REVIEW

**Comment by hkvision** [comment]
Don't forget to change the unit tests/examples and docs.

**Comment by xiejinglei** [comment]
> Don't forget to change the unit tests/examples and docs.

OK I am working on this now

**Review by hkvision — APPROVED** [review]

**Comment by yangw1234** [comment]
LGTM

**Comment by xiejinglei** [comment]
PR validation results are here:
http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Validation/5077/
http://10.239.47.210:18888/job/ZOO-PR-Python-ExampleTests/2292/

File: pyzoo/zoo/orca/learn/pytorch/torch_runner.py [hunk 1]
```diff
@@ -249,14 +249,13 @@ def should_wrap_dataloader(loader):
     def train_epochs(self, data_creator, epochs=1, batch_size=32, profile=False,
                      info=None, wrap_dataloader=None):
         config = self.config.copy()
-        if "batch_size" not in config:
-            config["batch_size"] = batch_size
+        config["batch_size"] = batch_size
```
Comment (hkvision) [inline pyzoo/zoo/orca/learn/pytorch/torch_runner.py:252]: do we still need to set batch_size in config?

File: pyzoo/zoo/orca/learn/pytorch/torch_runner.py [hunk 2]
```diff
@@ -299,17 +298,16 @@ def validate(self, data_creator, batch_size=32, num_steps=None, profile=False,
                  info=None, wrap_dataloader=None):
         """Evaluates the model on the validation data set."""
         config = self.config.copy()
-        if "batch_size" not in config:
-            config["batch_size"] = batch_size
+        config["batch_size"] = batch_size
```
Comment (hkvision) [inline pyzoo/zoo/orca/learn/pytorch/torch_runner.py:301]: same as above

Comment (xiejinglei) [reply pyzoo/zoo/orca/learn/pytorch/torch_runner.py:301]: That makes sense.. I will try to run validation again without that `config["batch_size"]`.

File: pyzoo/zoo/orca/learn/pytorch/torch_runner.py [hunk 3]
```diff
@@ -328,8 +324,7 @@ def validate(self, data_creator, batch_size=32, num_steps=None, profile=False,
     def predict(self, data_creator, batch_size=32, profile=False):
         """Evaluates the model on the validation data set."""
         config = self.config.copy()
-        if "batch_size" not in config:
-            config["batch_size"] = batch_size
+        config["batch_size"] = batch_size
```
Comment (hkvision) [inline pyzoo/zoo/orca/learn/pytorch/torch_runner.py:327]: also need to change predict?

Comment (xiejinglei) [reply pyzoo/zoo/orca/learn/pytorch/torch_runner.py]: Fixed now

**Comment by xiejinglei** [comment]
Update: `config["batch_size"] = batch_size` removed and `predict` function updated.
PR validation: http://10.239.47.210:18888/job/ZOO-PR-Validation/5082/

File: pyzoo/zoo/orca/learn/pytorch/estimator.py [hunk 4]
```diff
@@ -87,6 +87,11 @@ def __init__(self,
                  use_tqdm=False,
                  backend="torch_distributed",
                  workers_per_node=1):
+
+        if config is not None and "batch_size" in config:
+            raise Exception("Please do not specify batch_size in config. Input batch_size in the"
+                            " fit/evaluate function of the estimator instead.")
```
Comment (hkvision) [inline pyzoo/zoo/orca/learn/pytorch/estimator.py:93]: fit/evaluate/predict?

Comment (xiejinglei) [reply pyzoo/zoo/orca/learn/pytorch/estimator.py]: OK. Updated now

**Comment by jason-dai** [comment]
Also need to fix https://github.com/intel-analytics/analytics-zoo/blob/master/readthedocs/source/doc/Orca/Overview/data-parallel-processing.md#11-data-creator-function and https://github.com/intel-analytics/analytics-zoo/blob/master/readthedocs/source/doc/Orca/Overview/distributed-training-inference.md @xiejinglei

**Comment by xiejinglei** [comment]
> Also need to fix https://github.com/intel-analytics/analytics-zoo/blob/master/readthedocs/source/doc/Orca/Overview/data-parallel-processing.md#11-data-creator-function and https://github.com/intel-analytics/analytics-zoo/blob/master/readthedocs/source/doc/Orca/Overview/distributed-training-inference.md @xiejinglei

Thanks for pointing that out. I have created PR #3525 to fix the issues.

**Pull request merged**